platform/packages/currency/src/nls.rs

```diff
@@ -26,18 +29,45 @@ impl Currency for NlsPlatform {
     const DECIMAL_DIGITS: u8 = 6;
 }
 
-#[derive(Deserialize, PartialEq, Eq)]
-#[cfg_attr(any(test, feature = "testing"), derive(Debug))]
-pub struct Native {}
+impl Currency for NlsPlatform {
+    type Group = Native;
+}
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub struct Native(TypeId);
+
+impl MemberOf<Self> for Native {}
```

Comment:
I did it but it conflicted with another blanket impl:
```
impl<G, C> MemberOf<G> for C
where
    C: Currency,
    C::Group: MemberOf<G>,
```

Reply:
any idea how to overcome it?

Reply:
If overlapping empty trait implementations, negative trait bounds or negative equality operator in `where` clauses were allowed, we could've. Best we can do is require it to implement the trait.
* ```
  trait Currency {
      type Group: Group where Self: MemberOf<Group>;
  }
  ```
  OR
* ```
  trait Currency where Self: MemberOf<Group> {
      type Group: Group;
  }
  ```

Reply:
I don't know how the extra trait boundary solves the conflicting implementations. We should select which one to keep and which to implement manually. I have chosen to keep the impl MemberOf for C, currencies, since they are more than the groups.